src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATE_CLONE = Object.assign({}, state);
+  const RESULT_ARR = [];
+
+  for (const action of actions) {
+    const ACTION_TYPE = action.type;
+
+    switch (ACTION_TYPE) {
+      case 'addProperties':
+        addProperties(STATE_CLONE, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(STATE_CLONE, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(STATE_CLONE);
+        break;
+    }
+
+    RESULT_ARR.push({ ...STATE_CLONE });
+  }
+
+  return RESULT_ARR;
+
+  function addProperties(changingObject, extraData) {
+    Object.assign(changingObject, extraData);
+  }
+
+  function removeProperties(changingObject, keysToRemove) {
+    for (const key of keysToRemove) {
+      delete changingObject[key];
+    }
+  }
+
+  function clearProperties(changingObject) {
```

Comment:
as for a separate function with such name, it's fine logic
but in this particular case you might be able to do it:

![image](https://github.com/mate-academy/js_stateful-clones/assets/86923089/f972adb5-8a85-48c7-b3d1-adadd7f46f0c)

because you anyway do copy after `swith` block